server/repository/src/database_settings.rs

```diff
@@ -111,8 +111,52 @@ impl diesel::r2d2::CustomizeConnection<diesel::SqliteConnection, diesel::r2d2::E
 pub fn get_storage_connection_manager(settings: &DatabaseSettings) -> StorageConnectionManager {
     let connection_manager =
         ConnectionManager::<DBBackendConnection>::new(&settings.connection_string());
-    let pool = Pool::new(connection_manager).expect("Failed to connect to database");
-    StorageConnectionManager::new(pool)
+
+    // Note: the build() call isn't failing when you have an incorrect server or database name
+    // which is why the connect() call is made
+    match connection_manager.connect() {
+        Ok(_conn) => {
+            // the min_idle is a workaround to get pg running reliably on windows.
+            // if you don't require it any more, swap the pool creation lines out for this one:
+            // let pool = Pool::new(connection_manager).expect("Failed to connect to database");
+            let pool = Pool::builder()
+                .min_idle(Some(0))
+                .build(connection_manager)
+                .expect("Failed to connect to database");
+            StorageConnectionManager::new(pool)
+        }
+        Err(e) => {
+            if e.to_string().contains(
+                format!("database \"{}\" does not exist", &settings.database_name).as_str(),
```

Comment:
Might be nice to log something here, so the logs show that the app is trying to create a new database.